### ISSUE TITLE
gcc missing in  openwhisk/dockerskeleton

### ISSUE THREAD

**Issue by iainhouston** [openwhisk/openwhisk#1204]
## Environment details:

* local deployment, vagrant, native ubuntu, Mac OS, Bluemix, ...   
* version of docker, vagrant, ubuntu, ...

## Steps to reproduce the issue:

1.   `./buildAndPush.sh my_namespace/skeleton`

## Provide the expected results and outputs:

```
...
latest: digest: sha256:...
...
```

## Provide the actual results and outputs:

```
$ ./buildAndPush.sh iainh/skeleton
Using iainh/skeleton as the image name
Sending build context to Docker daemon 6.656 kB
Step 1 : FROM openwhisk/dockerskeleton
latest: Pulling from openwhisk/dockerskeleton
e110a4a17941: Pull complete 
4b8a645b1f9e: Pull complete 
30921c0ad717: Pull complete 
bddbfbdffbff: Pull complete 
de0e3cfef544: Pull complete 
e1cca3827cc8: Pull complete 
6dd273025a52: Pull complete 
a6fd5c79e05c: Pull complete 
7efedd5ac033: Pull complete 
Digest: sha256:7670e717d00a1713146ce88a006b4b760e6a3aebf11e6fe9369eeeb383bddd3e
Status: Downloaded newer image for openwhisk/dockerskeleton:latest
 ---> c669352ac080
Step 2 : ENV DEBIAN_FRONTEND noninteractive
 ---> Running in 3d3a8170267e
 ---> 5b3e25422290
Removing intermediate container 3d3a8170267e
Step 3 : ENV FLASK_PROXY_PORT 8080
 ---> Running in 3165dfab51e6
 ---> e211c62b8e36
Removing intermediate container 3165dfab51e6
Step 4 : ADD example.c /action/example.c
 ---> 1013122d9d10
Removing intermediate container e2af339fa232
Step 5 : RUN cd /action; gcc -o exec example.c
 ---> Running in 65d07d734a90
/bin/sh: gcc: not found
The command '/bin/sh -c cd /action; gcc -o exec example.c' returned a non-zero code: 127
Docker build failed
```

**Comment by rabbah** [comment]
@iainhouston bug confirmed - we introduced a second dockerfile for deployment reasons and it's untested, and hence broken.

As a workaround, you can replace the Dockerfile in the `/dockerSkeleton` directory with:

```
# Dockerfile for example whisk docker action
FROM openwhisk/dockerskeleton
 
ENV FLASK_PROXY_PORT 8080

### Add source file(s)
ADD example.c /action/example.c

RUN apk add --no-cache --virtual .build-deps \
        bzip2-dev \
        gcc \
        libc-dev \
### Compile source file(s)
 && cd /action; gcc -o exec example.c \
 && apk del .build-deps

CMD ["/bin/bash", "-c", "cd actionProxy && python -u actionproxy.py"]
```

**Comment by rabbah** [comment]
Proposed patch: https://github.com/openwhisk/openwhisk/compare/master...rabbah:issue-1204?expand=1.

**Comment by rabbah** [comment]
Closed by #1205.
